go-controller/observability-lib/sampledecoder/nb_client.go

```diff
@@ -0,0 +1,110 @@
+package sampledecoder
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/model"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/observability-lib/ovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"k8s.io/klog/v2/textlogger"
+)
+
+const OVSDBTimeout = 100 * time.Second
```

Comment:
100 seconds seems a bit too long.  For other DB clients, e.g., `libovsdb.newClient()` we use 10 seconds.

Also, why not use the libovsdb.newClient() directly?  It would allow support for remote databases too (see the non-IC comment I made earlier).

Reply:
that was an attempt to decrease the amount of dependencies for this lib as asked by the observ team, since libovsdb.go has quite some imports that were not used at the time.
Now when we know we want to support remote-db connection, I probably will end up using it directly at some point.
About timeout - we updated it recently in ovn-k to default to 100 seconds, but that is mostly for large db transactions, that are not expected with the lib, so I can make it 10 seconds.